sql/item_func.cc

```diff
@@ -7504,14 +7504,21 @@ static bool check_table_and_trigger_access(Item **args, bool check_trigger_acl,
   }
 
   // Make sure we have safe string to access.
-  schema_name_ptr->c_ptr_safe();
-  table_name_ptr->c_ptr_safe();
+  const char *sch_name = schema_name_ptr->c_ptr_safe();
+  const char *tbl_name = table_name_ptr->c_ptr_safe();
 
   // Check if table is hidden.
   THD *thd = current_thd;
   if (is_hidden_by_ndb(thd, schema_name_ptr, table_name_ptr))
     DBUG_RETURN(false);
 
+  if ((strcmp(sch_name, "mysql") == 0 &&
```

Comment:
To hide compression dictionary tables from "SHOW TABLES". This is needed because mysqlcheck tries to access tables listed by "SHOW TABLES" and when it tries to do CHECK TABLE mysql.compression_dictionary table, it gets "access denied error".

So in short, it should be treated as hardcoded dictionary tables and hide them from SHOW TABLES.

Reply:
Please add a comment that these need to be hidden explicitly like DD tables while they are not technically DD tables